text/0000-decouple-dependencies.md

```diff
@@ -0,0 +1,403 @@
+
+For example:
+```toml
+[[metadata.validations]]
+dependency-id = "com.example.jre"
+supported = [ "8.0.*", "11.0.*", "17.0.*" ]
+
+[[metadata.validations]]
+dependency-id = "com.example.nodejs"
+supported = [ "^16.0", "^17.0", "^18.0" ]
+
+[[metadata.validations]]
+dependency-id = "com.example.tomcat"
+supported = [ "8\.5\.\d+", "9\.0\.\d+", "10\.0\.\d+" ]
+type = “regex”
+```
+
+A buildpack is encouraged to be as permissive as possible. This ensures that a
+buildpack author won’t have to frequently update this metadata. This should be
+balanced by buildpack authors to provide compatibility guarantees with the
+tools required to run the buildpack and for the software it is installing to
+run. Generally, we believe that most buildpacks will be compatible with the
+major versions of software they presently support and that new major versions
+of dependencies should be tested and validations should be expanded after
+testing.
+
+## Metadata Distribution
+There is no prescribed method for distributing metadata. It could be done in a
+variety of ways, including HTTPS/SFTP distributed archives, `rsync` of remote
+directories, or even distributed as an image through an image registry.
+
+For the Paketo project, this proposal suggests distributing metadata through
+images in an image registry. This allows the project to use the existing image
+registry to distribute the metadata. Image registries also have inherent
+properties that help with security, like an image cannot be modified without
+creating a new hash for the image and images can be signed (signing is out of
+scope for this proposal). In addition, images are easily versioned such that
+users can hold back updates to dependencies if desired and are easily cached.
+
+The contents of the image will contain the directory structure defined in
+[Metadata Format](#metadata-format). There should not be a top-level directory
+added, so the root of the image should contain all of the directories created
+as top-level organization names, like `com` or `org`. All of the metadata is to
+be included in a single layer. Updates to metadata will require downloading the
+entire layer again, however, it is a single layer and the size is expected to
+be small so this should be very fast.
+
+Further, this proposal suggests subdividing metadata images by project
+sub-team. Each sub-team will be given a unique reverse domain name like
+`io.paketo.java` and `io.paketo.utilities`. In this way, the project’s metadata
+can be easily combined without having conflicts.
+
+This allows users to pick and choose the metadata that’s relevant to their
+needs.
+
+## Buildpack Dependency Metadata Interface
+The interface between dependency metadata and a buildpack is simple. A single
+directory of metadata will be presented to the buildpack. It will be presented
+at the location specified by `BP_DEPENDENCY_METADATA`, which defaults to
+`/platform/deps/metadata` (the intent is to pilot and try out this or possibly
```

Comment:
I'd rather keep it as a default because then the user doesn't have to provide it at build time. I might be wrong, but I don't think you could set the env var in an upstream buildpack because the env vars do not propagate during the `Detect` phase, and generally the downstream buildpack would have to know where the metadata is during `Detect` as well  as `Build`.

Reply:
Would it need to know about metadata during detect though? Do you have something specific in mind?